doc/cprover-manual/function-contracts.md

```diff
@@ -0,0 +1,502 @@
+# Function Contracts
```

Comment:
Overall comment: this file contains too much "how" and not enough "what".  We do not need to explain the implementations of ASSIGNS and is_fresh, but we should explain what they do, perhaps using examples.  Also, my thought is that there should be a page for each construct (similar to a reference manual) with examples that demonstrate how the construct works.  So, I'd like to see some refactoring, and a few more examples of how the constructs work.

Reply:
I've re-write and re-sctructured the documentation to highlight "what" instead of "how" and added more examples.